### ISSUE TITLE
UnusedPrivateMember rule counts package declarations as "usage references"

### ISSUE THREAD

**Issue by luanpotter** [detekt/detekt#5934]
## Expected Behavior

The following code:

```kotlin
package xyz.luan.test

fun main() {
    val xyz = "a"
    val luan = "b"
    val test = "c"
    println("Hello, world!")
}
```

Should fail detekt validation of the `UnusedPrivateMember` for the 3 unused local variables. Note that a similar snippet does fail:

```kotlin
package xyz.luan.test

fun main() {
    val xyz2 = "a"
    val luan = "b"
    val test = "c"
    println("Hello, world!")
}
```

Yields:

```
> Task :detekt FAILED
detekt-testbed/src/main/kotlin/xyz/luan/test/Main.kt:7:9: Private property `xyz2` is unused. [UnusedPrivateMember]
```

## Observed Behavior

The code is not reported and detekt succeeds even though there is an unused local variable.

Note: the same issue can happen with other types of unused private properties, like fields or parameters.

Any "bits" composing the package declaration are considered as variable uses. When in fact, package declarations should not count for referencing variables at all.

## Steps to Reproduce

Please see [my repo](https://github.com/luanpotter/detekt-testbed/tree/luan.unused-private-property-improvements) for a fully reproducible example (just run `./gradlew detekt`). Or run detekt on the code snippets above.  Should be simple to reproduce. I am using version 1.22.0.

## Context

I noticed that I had some private constructor parameters not used on some classes that weren't flagged, and they all had the same name. I had many wrong theories but did some science and turns out it was the package name.

The fix should be fairly simple, I will link a PR to this issue with a proposal.

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used: 8.0.2
* Gradle scan link: https://scans.gradle.com/s/ne6pd7y7cf3t2
* Operating System and version: Arch Linux (kernel 22.3.0)
* Link to your project: https://github.com/luanpotter/detekt-testbed/tree/luan.unused-private-property-improvements
